experimental/dds/tree2/src/test/shared-tree-core/branch.spec.ts

```diff
@@ -439,6 +435,54 @@ describe("Branches", () => {
 		assert.equal(branch.getHead().revision, nullRevisionTag);
 	});
 
+	describe("all nested forks and transactions are disposed and aborted when transaction is", () => {
```

Comment:
Since you added logic to deal with the fact that transactions can be nested, you should add tests to verify that the added logic is funtionning as intended.